zebra-consensus/src/verify.rs

```diff
@@ -0,0 +1,60 @@
+//! Context-dependent block and transaction verification for Zebra.
+//!
+//! Verification occurs in two stages:
+//!   - context-free verification of signatures, proofs, and scripts, handled by `zebra-chain` and
+//!     `zebra-script`
+//!   - context-dependent verification of the chain, handled by `zebra-consensus` (this module)
+//!
+//! Verification is provided via a `tower::Service`, to support batching.
+
+use std::sync::Arc;
+use zebra_chain::block::Block;
+use zebra_chain::transaction::Transaction;
+
+/// A context-dependent block or transaction verification request.
+///
+/// This has a `Flush` variant, to allow manually flushing queued verification
+/// requests, even when the batching service is wrapped in other `tower` layers.
+#[derive(Debug)]
+// The associated data doesn't need separate documentation.
+#[allow(missing_docs)]
```

Comment:
I want to add `!#[deny(missing_docs)]` to this module. Then I think this allow will be required.

Reply:
ooh, I get this now. I didn't realize that it would try to force you to document every data member as well, this all makes a lot more sense. I think we should just update the wording to be a little more clear, `associated data` makes me think traits and associated types.